### DISCUSSION TITLE
[Native DLL] Pointer to pointer to structs

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#3026] [General]
I think pointer to pointer to structs is meaningless. I think it should be array of pointer to structs, that in the world of C has the same meaning of array of objects in OOP. The problem is, I don't know how to handle it in my translation. Should I simply use `IntPtr`? But if I do so, how could I pass the data to it?

**Comment by SunSerega** [comment]
In most cases, the `external` subprogram would probably take an `array of IntPtr`. Then non-`external` subprogram should manually marshal each item of the original array to `IntPtr` (allocated using `Marshal.AllocHGlobal`)

But everything from https://github.com/pascalabcnet/pascalabcnet/discussions/3025 still applies. `my_struct** v` doesn't mean you need to pass in an array.

Maybe this is a function that returns an `array of my_struct`, and stores it at the address you passed in as `v`?
In that case `var v: IntPtr` makes more sense.
